HW05_Xingruo_Zhang.Rmd

```diff
@@ -0,0 +1,121 @@
+---
+title: "HW05_Xingruo_Zhang"
+author: "Summer Zhang"
+date: "8/2/2020"
+output: github_document
+---
+
+
+```{r, message = FALSE}
+library(tidyverse)
+library(readr)
+library(knitr)
+library(stringr)
+
+dad_mom <- read_delim("dad_mom.txt", delim = "\t")
+
+auc_1 <- read_csv("CTRP_files/AUC_1.csv")
+auc_2 <- read_csv("CTRP_files/AUC_2.csv")
+cancer_cell_line_info <- read_csv("CTRP_files/cancer_cell_line_info.csv")
+compound_info <- read_csv("CTRP_files/compound_info.csv")
+Experiment_info <- read_csv("CTRP_files/Experiment_info.csv")
+```
+
+## 1. Tidying the dad_mom file
+```{r}
+dad_mom2 <- dad_mom %>% 
+  # Combine parents' names into one column instead of two
+  gather(key = "parent", value = "name_parent", name_dad, name_mom) %>% 
+  # Correct values in the parent column
+  mutate(parent = ifelse(parent == "name_dad", "dad", "mom")) %>% 
+  # Assign correct income to each parent
+  mutate("income_parent" = ifelse(parent == "dad", income_dad, income_mom)) %>% 
```

Comment:
This is a very effective approach to reshape the data set! I learnt from someone else's code that we can also use the combination of gather( ), separate( ), and spread( ) to achieve the same result.